libfwupdplugin/fu-cabinet.c

```diff
@@ -453,6 +454,33 @@ fu_cabinet_set_container_checksum_cb (XbBuilderFixup *builder_fixup,
 	return TRUE;
 }
 
+static gboolean
+fu_cabinet_set_lowercase_checksum_cb(XbBuilderFixup *builder_fixup,
+				     XbBuilderNode *bn,
+				     gpointer user_data,
+				     GError **error)
+{
+	g_autoptr(XbBuilderNode) csum = NULL;
+
+	/* verify it exists */
+	if (g_strcmp0(xb_builder_node_get_element(bn), "artifact") == 0)
+		/* don't care whether it's sha256, sha1 or something else so don't check for
+		 * specific value */
+		csum = _xb_builder_node_get_child_by_element_attr(bn, "checksum", "type", NULL);
```

Comment:
Isn't this just fixing up the "first" child? What if md5 was first listed?

Reply:
Adjusted it again.